### PR TITLE
Throttle policy export import support for APIM

### PR DESCRIPTION
**Pull request by RansikaG**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: import-export-cli/Subscription-Policies/AsyncBronze.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (rmsamitha) [inline import-export-cli/Subscription-Policies/AsyncBronze.json:1]: Why was it required to commit these all the policy files newly?

File: apim-traffic-tool/README.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ Some of the default configurations for WSO2 API Manager and default scenario are
 
  > It is recommended to add `setenv.sh` script in the `<TOOL_HOME>/resources/add-on` directory to the `<JMETER_HOME>/bin` directory in order to increase the heap size of the JMeter if you are simulating DOS or DDOS attacks with heavy concurrency.
 
-7. Add swagger definitions for all APIs as separate json files in the `<TOOL_HOME>/data/swagger/` directory. Filename should be equal to `<api_name>.json` in lower case letters.
+7. Add swagger definitions for all APIs as separate json files in the `<TOOL_HOME>/data/swagger/` directory. fileName should be equal to `<api_name>.json` in lower case letters.
```
[the remark this answers is not in the document]
Comment (wasuradananjith) [reply apim-traffic-tool/README.md:140]: Seems a simple string replace has changed the occurrences. @RansikaG shall we revert these file changes? 
59 files have been changed in this PR. Shall we keep only the files you change and revert the rest of the files?

File: import-export-cli/api_meta.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+deploy:
```
Comment (rmsamitha) [inline import-export-cli/api_meta.yaml:1]: What is this file added newly ?

File: import-export-cli/Subscription-Policies/AsyncBronze.json (hunk 1 above)
Comment (wasuradananjith) [inline import-export-cli/Subscription-Policies/AsyncBronze.json:1]: This is not related to code I guess. A test artifact that you exported?
There are several policy files like this below. Let's remove all of them.

File: import-export-cli/api_meta.yaml (hunk 3 above)
Comment (wasuradananjith) [inline import-export-cli/api_meta.yaml:1]: This file is also not from the code. Isn't it?

File: import-export-cli/cmd/deprecated/exportAPI.go [hunk 4]
```diff
@@ -61,7 +61,7 @@ var ExportAPICmdDeprecated = &cobra.Command{
 		utils.Logln(utils.LogPrefixInfo + exportAPICmdLiteral + " called")
 		var apisExportDirectory = filepath.Join(utils.ExportDirectory, utils.ExportedApisDirName)
 
-		cred, err := cmd.GetCredentials(cmd.CmdExportEnvironment)
+		cred, err := cmd.GetCredentials(cmd.ExportEnvironment)
```
Comment (rmsamitha) [inline import-export-cli/cmd/deprecated/exportAPI.go:64]: Why was this changed?

File: unnamed.patch [hunk 5]
```diff
@@ -0,0 +1,975 @@
+Index: import-export-cli/impl/common.go
```
Comment (SanojPunchihewa) [inline unnamed.patch:1]: Shall we remove this file?

File: import-export-cli/cmd/deprecated/exportAPI.go (hunk 4 above)
Comment (rmsamitha) [reply import-export-cli/cmd/deprecated/exportAPI.go:64]: Check all the places changed

Comment (wasuradananjith) [inline import-export-cli/cmd/deprecated/exportAPI.go:64]: We should not rename this variable. I saw that this variable was removed from the root.go. We should not do that. 

Since your command space is using ExportEnvironment, try using CmdExportEnvironment. Since it is in root, it is accessible to all the child go files.

File: import-export-cli/cmd/getPolicies.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+// GetPoliciesCmd  represents the get command for policies
+var GetPoliciesCmd = &cobra.Command{
+	Use:     GetPoliciesCmdLiteral,
+	Short:   GetPoliciesCmdShortDesc,
+	Long:    GetPoliciesCmdLongDesc,
+	Example: GetPoliciesCmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + GetCmdLiteral + " called")
+
```
Comment (rmsamitha) [inline import-export-cli/cmd/getPolicies.go:39]: Remove empty lines

File: import-export-cli/docs/apictl_set.md [hunk 7]
```diff
@@ -34,7 +34,7 @@ apictl set api-logging --api-id bf36ca3a-0332-49ba-abce-e9992228ae06 --log-level
 ### Options
 
 ```
-      --export-directory string           Path to directory where APIs should be saved (default "/Users/wso2user/.wso2apictl/exported")
+      --export-directory string           Path to directory where APIs should be saved (default "/home/ransika/.wso2apictl/exported")
```
Comment (wasuradananjith) [inline import-export-cli/docs/apictl_set.md:37]: Let's revert this. (This is a limitation in our do generating tool)

File: import-export-cli/cmd/importPolicy.go [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+// ImportPolicyCmd represents the import command for a policy
+var ImportPolicyCmd = &cobra.Command{
+	Use:     ImportPolicyCmdLiteral,
+	Short:   ImportPolicyCmdShortDesc,
+	Long:    ImportPolicyCmdLongDesc,
+	Example: ImportPolicyCmdExamples,
+	Run: func(cmd *cobra.Command, args []string) {
+		utils.Logln(utils.LogPrefixInfo + ExportCmdLiteral + " called")
+
```
Comment (rmsamitha) [inline import-export-cli/cmd/importPolicy.go:39]: Remove empty lines

File: import-export-cli/impl/common.go [hunk 9]
```diff
@@ -97,6 +98,7 @@ func IncludeMetaFileToZip(sourceZipFile, targetZipFile, metaFile string, metaDat
 
 	err = utils.Zip(tmpClonedLoc, targetZipFile)
 	if err != nil {
+		fmt.Println("This is it")
```
Comment (wasuradananjith) [inline import-export-cli/impl/common.go:101]: Why do we have this?

File: import-export-cli/cmd/root.go [hunk 10]
```diff
@@ -47,7 +47,6 @@ var cfgFile string
 var insecure bool
 var cmdPassword string
 var CmdUsername string
-var CmdExportEnvironment string
```
Comment (rmsamitha) [inline import-export-cli/cmd/root.go:50]: Reason to remove?

File: import-export-cli/impl/importThrottlePolicy.go [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+	"os"
+	"strconv"
+)
+
+func ImportThrottlingPolicyToEnv(accessOAuthToken, importEnvironment, importThrottlingPolicyFile string, importThrottlePolicyUpdate bool) error {
+	adminEndpoint := utils.GetAdminEndpointOfEnv(importEnvironment, utils.MainConfigFilePath)
+	return ImportThrottlingPolicy(accessOAuthToken, adminEndpoint, importThrottlingPolicyFile, importThrottlePolicyUpdate)
+}
+
+func ImportThrottlingPolicy(accessOAuthToken, adminEndpoint, importPath string, importThrottlePolicyUpdate bool) error {
```
Comment (wasuradananjith) [inline import-export-cli/impl/importThrottlePolicy.go:36]: We don't need this function, I think. You can embed the implementation of this to the function **importThrottlingPolicy**, and call it directly from **ImportThrottlingPolicyToEnv**, isn't it?

File: import-export-cli/impl/common.go (hunk 9 above)
Comment (rmsamitha) [inline import-export-cli/impl/common.go:101]: Is this a testing line? Please remove

File: import-export-cli/impl/common.go [hunk 12]
```diff
@@ -19,6 +19,7 @@
 package impl
 
 import (
+	"fmt"
```
Comment (rmsamitha) [inline import-export-cli/impl/common.go:22]: Remove this too

File: import-export-cli/docs/apictl_set.md (hunk 7 above)
Comment (rmsamitha) [inline import-export-cli/docs/apictl_set.md:37]: This should not be changed

File: import-export-cli/cmd/root.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (RansikaG) [reply import-export-cli/cmd/root.go]: Fixed

File: import-export-cli/impl/common.go (hunk 9 above)
Comment (RansikaG) [reply import-export-cli/impl/common.go]: Removed

Comment (RansikaG) [reply import-export-cli/impl/common.go]: removed

File: apim-traffic-tool/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (RansikaG) [reply apim-traffic-tool/README.md]: Reverted the file change

File: import-export-cli/Subscription-Policies/AsyncBronze.json (hunk 1 above)
Comment (RansikaG) [reply import-export-cli/Subscription-Policies/AsyncBronze.json]: Removed those artifacts

Comment (RansikaG) [reply import-export-cli/Subscription-Policies/AsyncBronze.json]: Removed all the artifacts

File: import-export-cli/api_meta.yaml (hunk 3 above)
Comment (RansikaG) [reply import-export-cli/api_meta.yaml]: removed the file

Comment (RansikaG) [reply import-export-cli/api_meta.yaml]: yes, it was removed

File: import-export-cli/cmd/deprecated/exportAPI.go (hunk 4 above)
Comment (RansikaG) [reply import-export-cli/cmd/deprecated/exportAPI.go]: All the places changed

File: import-export-cli/cmd/root.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (RansikaG) [reply import-export-cli/cmd/root.go]: Added again

File: import-export-cli/cmd/deprecated/exportAPI.go (hunk 4 above)
Comment (RansikaG) [reply import-export-cli/cmd/deprecated/exportAPI.go]: restored it to the previous state

File: import-export-cli/cmd/getPolicies.go (hunk 6 above)
Comment (RansikaG) [reply import-export-cli/cmd/getPolicies.go]: removed empty lines

File: import-export-cli/cmd/root.go (hunk 10 above)
Comment (RansikaG) [reply import-export-cli/cmd/root.go]: Restored it

File: import-export-cli/cmd/importPolicy.go (hunk 8 above)
Comment (RansikaG) [reply import-export-cli/cmd/importPolicy.go]: Removed empty lines

File: import-export-cli/impl/common.go (hunk 12 above)
Comment (RansikaG) [reply import-export-cli/impl/common.go]: removed

File: unnamed.patch (hunk 5 above)
Comment (RansikaG) [reply unnamed.patch]: Removed the file

**Review by rmsamitha — APPROVED** [review]
